diem-node/src/lib.rs

```diff
@@ -387,11 +388,13 @@ pub fn setup_environment(node_config: &NodeConfig, logger: Option<Arc<Logger>>)
     // and pass network handles to mempool/state sync
 
     // for state sync to send requests to mempool
-    let (state_sync_to_mempool_sender, state_sync_requests) =
-        channel(INTRA_NODE_CHANNEL_BUFFER_SIZE);
+    let mut mempool_notifier = MempoolNotifier::new();
+    let mempool_receiver = mempool_notifier
+        .take_notification_receiver()
+        .expect("Unable to take the mempool receiver!");
```

Comment:
Maybe we just make `MempoolNotifier::new` return the `notifier` and the `receiver` like a `channel`?